### PR TITLE
👷 [RUMF-1441] upgrade yarn

### PR DESCRIPTION
**Pull request by BenoitZugmeyer**
## Motivation

I can't merge https://github.com/DataDog/browser-sdk/pull/2169 because `yarn` v1 has an issue and somehow installs multiple versions of `wrap-ansi` in the same `node_modules/...` directory. To reproduce:

```sh
$ rm -r node_modules yarn.lock
$ yarn
$ yarn build     # this works
$ rm -r node_modules
$ yarn
$ yarn build   # this fails:
/.../browser-sdk/node_modules/cliui/build/index.cjs:293
const wrap = require('wrap-ansi');
             ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /.../browser-sdk/node_modules/cliui/build/index.cjs not supported.
Instead change the require of index.js in /.../browser-sdk/node_modules/cliui/build/index.cjs to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/.../browser-sdk/node_modules/cliui/build/index.cjs:293:14)
    at Object.<anonymous> (/.../browser-sdk/node_modules/yargs/build/index.cjs:2861:12)
    at Object.<anonymous> (/.../browser-sdk/node_modules/yargs/yargs:3:69)
    at libs/core/src/lib/cli.ts (/.../browser-sdk/node_modules/lerna/dist/cli.js:253:28)
    at __init (/.../browser-sdk/node_modules/lerna/dist/cli.js:11:56)
    at libs/core/src/index.ts (/.../browser-sdk/node_modules/lerna/dist/cli.js:4651:5)
    at __init (/.../browser-sdk/node_modules/lerna/dist/cli.js:11:56)
    at packages/lerna/src/index.ts (/.../browser-sdk/node_modules/lerna/dist/cli.js:8897:5)
    at __require (/.../browser-sdk/node_modules/lerna/dist/cli.js:14:50)
    at Object.<anonymous> (/.../browser-sdk/node_modules/lerna/dist/cli.js:8929:3) {
  code: 'ERR_REQUIRE_ESM'
}
```

As it turns out, invoking `yarn` to install dependencies without a `yarn.lock` and with a freshly generated `yarn.lock` does not produce the same `node_modules` output.

## Changes

* Upgrade yarn to 3.5.1
* Make the necessary adjustments to work around breaking changes
* Update the CI cache to make it faster (see commit message https://github.com/DataDog/browser-sdk/pull/2199/commits/33b6a7ba0b67a40776417e9c1badcf6d07f3136b)

## Testing

- [x] Local
- [ ] Staging
- [ ] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: test/app/.yarnrc.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+yarnPath: ../../.yarn/releases/yarn-3.5.1.cjs
```
Comment (yannickadam) [inline test/app/.yarnrc.yml:1]: ❓ We're using `pnp` for the test app ?

Comment (BenoitZugmeyer) [reply test/app/.yarnrc.yml:1]: Hum somehow the `nodeLinker` value is inherited from the root `package.json`

```
$ cd test/app
$ yarn config --why | grep nodeLinke
➤ YN0000: nodeLinker                    /Users/benoit.zugmeyer/go/src/github.com/DataDog/browser-sdk/.yarnrc.yml            'node-modules'
```

Comment (BenoitZugmeyer) [reply test/app/.yarnrc.yml]: I removed `test/app/.yarnrc.yml` as it turns out everything is inherited correctly

Comment (yannickadam) [reply test/app/.yarnrc.yml]: 👍

**Review by yannickadam — APPROVED** [review]

**Review by bcaudan — APPROVED** [review]
LGTM

**Pull request merged**